src/domain/dataset-update-flow/src/aggregates/update.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{DateTime, Utc};
+use event_sourcing::*;
+use kamu_task_system::TaskID;
+use opendatafabric::DatasetID;
+
+use crate::*;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Aggregate, Debug)]
+pub struct Update(Aggregate<UpdateState, (dyn UpdateEventStore + 'static)>);
```

Comment:
# Updates and Triggers
`retrigger()` and `UpdateSecondaryTrigger` - can you explain the idea behind this? 

Can there be a tertiary trigger?

Is this for cases when e.g. an `AutoPolling` update was scheduled, and then user clicks "Refresh" button ... then `Manual` will become a secondary trigger?

In this case does it make sense to have a `Vec<UpdateTrigger>` in the state? e.g. in case when one derivative dataset accumulates 3 `InputDataset` triggers while being throttled?

I don't see any `UpdateTrigger`s in the `UpdateState` struct currently btw.

Reply:
The idea is that event store of the `DatasetUpdate` may be used to draw provenance records directly. It does not change the state of the aggregate, as it's not getting rescheduled or something, but just forms the detailed history.

Secondary triggers could be:
 - refresh manually when auto polling triggered (like you suggested)
 - derived dataset re-triggered from one or more inputs as it's still waiting for scheduling

Reply:
I think I will add triggers and start conditions to the state model as soon as this will be required for scheduling decisions. This part is not prototyped yet, so for now wasn't required.